### PR TITLE
Set assume_static_by_default to True in Dynamo config

### PR DESCRIPTION
**Pull request by gmagogsfm**
We expect fine grained dynamic shape enabled at all times, which means that a dimension is assumed to be static unless user explicitly says otherwise.

Differential Revision: D45473365

### CODE REVIEW

File: torch/_export/__init__.py [hunk 1]
```diff
@@ -108,6 +109,7 @@ def _export(
             tracing_mode="symbolic",
             decomposition_table=DECOMP_TABLE,
             constraints=constraints,
+            assume_static_by_default=True,
```
Comment (avikchaudhuri) [inline torch/_export/__init__.py:112]: I think this is the right thing to do, but why bother adding it to the export dynamo config in that case? In other words, regardless of whether the config is `False`, here you're making it `True`.

Comment (gmagogsfm) [reply torch/_export/__init__.py:112]: Because right now Dynamo's API surface on assume_static_by_default is a bit unclear, so I am setting it in both places to be safe.

Comment (avikchaudhuri) [reply torch/_export/__init__.py:112]: IMO it adds to the confusion.

Comment (gmagogsfm) [reply torch/_export/__init__.py:112]: @voznesenskym Michael, what's the plan for this flag? Is it going to be in the dynamo.export API or is it going to be in config.py or is it going to disappear?

Comment (avikchaudhuri) [reply torch/_export/__init__.py:112]: I think eventually it will disappear from `dynamo.config` (behaving as if `True`), at which point it should disappear everywhere. Until then it will probably stay, and I think the best intermediate behavior is either
(1) override to `True` always, as is here, but take it out of `ExportDynamoConfig`, or
(2) take an "or" of `ExportDynamoConfig.assume_static_by_default` and `dynamo.config.assume_static_by_default`.
In other words, intermediate behavior should be <= 2, eventual behavior will be 0.

Comment (gmagogsfm) [reply torch/_export/__init__.py:111]: Removed the logic to set dynamo.config, PTAL

**Review by avikchaudhuri — APPROVED** [review]
